src/coreneuron/io/reports/report_event.cpp

```diff
@@ -38,43 +41,78 @@ ReportEvent::ReportEvent(double dt,
 }
 
 void ReportEvent::summation_alu(NrnThread* nt) {
-    // Sum currents only on reporting steps
-    if (step > 0 && (static_cast<int>(step) % reporting_period) == 0) {
-        auto& summation_report = nt->summation_report_handler_->summation_reports_[report_path];
-        // Add currents of all variables in each segment
-        double sum = 0.0;
-        for (const auto& kv: summation_report.currents_) {
-            int segment_id = kv.first;
-            for (const auto& value: kv.second) {
-                double current_value = *value.first;
-                int scale = value.second;
-                sum += current_value * scale;
+    auto& summation_report = nt->summation_report_handler_->summation_reports_[report_path];
+    // Add currents of all variables in each segment
+    double sum = 0.0;
+    for (const auto& kv: summation_report.currents_) {
+        int segment_id = kv.first;
+        for (const auto& value: kv.second) {
+            double current_value = *value.first;
+            int scale = value.second;
+            sum += current_value * scale;
+        }
+        summation_report.summation_[segment_id] = sum;
+        sum = 0.0;
```

Comment:
another just nitpick?

```suggestion

    for (const auto& kv: summation_report.currents_) {
        double sum = 0.0;
        int segment_id = kv.first;
        for (const auto& value: kv.second) {
            double current_value = *value.first;
            int scale = value.second;
            sum += current_value * scale;
        }
        summation_report.summation_[segment_id] = sum;
```